### PR TITLE
Remove double slash from View on Site links

### PR DESCRIPTION
**Pull request by erino**

### CODE REVIEW

**Comment by bensymonds** [comment]
Does this cause an issue, or was it just annoying you too much?

I think the real issue here is the way `page_path` works. It's not great that we'd have to remember to do the `gsub` everywhere else. Is there any chance a tweak to the route would make it work? E.g. instead of

```ruby
get '*path' => 'pages#show', as: :page
```

do

```ruby
get '/*path' => 'pages#show', as: :page
```

?

**Comment by erino** [comment]
We're really only using this in the admin and yes the `gsub` isn't ideal. Changing the route doesn't make a difference as paths start with a `/`.

File: app/helpers/admin/admin_helper.rb [hunk 1]
```diff
@@ -59,4 +59,13 @@ def render_custom_links
   def cms_title
     @page.try(:name) || "Slices CMS"
   end
+
+  # Create a link to view the current page on the live site
+  #
+  # @!visibility private
+  def link_to_view_page(page, options = {})
+    options.reverse_merge!(target: 'site', class: 'view-page')
```
Comment (bensymonds) [inline app/helpers/admin/admin_helper.rb]: The old version had `target: '_blank'`, so is this change intentional? The commit is about removing the double slash so can't tell...

Same question about the `return unless page.active?` bit too. If we're changing other behaviour of the method, it would be nice to split into separate commits to keep it clear.

Comment (bensymonds) [inline app/helpers/admin/admin_helper.rb]: Also `options.reverse_merge!` mutates the passed-in argument, which seems a bit dangerous! The caller might re-use the `options` hash without realising it changed.

**Comment by erino** [comment]
The `target: '_blank'` and `return unless page.active?` where changed to harmonise the two link helpers. Maybe we should keep `_blank` rather then `site`

**Comment by bensymonds** [comment]
> the two link helpers

Which two? I can't see them. Not sure I understand the change either - if you visit the link and the page is not active, won't you just get a 404?

**Comment by erino** [comment]
Sorry rather then helpers I meant the two places where the CMS links to
front end (in the sitemap and whilst editing a page).

On 23 March 2015 at 09:45, Ben Symonds <notifications@github.com> wrote:

> the two link helpers
>
> Which two? I can't see them. Not sure I understand the change either - if
> you visit the link and the page is not active, won't you just get a 404?
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/withassociates/slices/pull/80#issuecomment-84916103>.
>



-- 
Erin Staniland
Director

- - - -

Pair With Associates
A problem shared is a problem solved
http://pair.withassociates.com/

With Associates
54B Downham Road, London, N1 5AS
+44 (0) 20 7923 4757 www.withassociates.com

- - - -

With Associates is the trading name of With Associates Digital Limited
Registered office: 54B Downham Road, N1 5AS
Registered in England under registration No. 08928808

**Comment by bensymonds** [comment]
Ok discussed the `page.active?` bit offline - let's go with it as is. So :+1:.
